src/coord/src/catalog.rs

```diff
@@ -450,20 +450,25 @@ impl CatalogEntryMap {
                 _ => {}
             }
         }
-        MultiWriteHandle::new_from_streams(handles.into_iter())
-            .ok()
-            .map(|write_handle| TablePersistMultiDetails {
+        if handles.len() > 0 {
+            let write_handle = MultiWriteHandle::new_from_streams(handles.into_iter());
+            Some(TablePersistMultiDetails {
                 all_table_ids,
                 write_handle,
             })
+        } else {
+            None
+        }
     }
 
     pub fn persist_multi_details(&self) -> Option<&TablePersistMultiDetails> {
+        // TODO: The handle is not Eq anymore because of the errors it can have. We could have a
+        // custom `Eq` impl that ignores the errors?
```

Comment:
this check (which is very important IMO, we should find a way to keep it) maybe only needs PartialEq? if that's right, then I think this is easy, if either side has an error then they're neither equal nor not equal. this would indeed me a custom impl

Reply:
That works!